### PR TITLE
add mmap option for faiss on disk indices

### PR DESCRIPTION
**Pull request by dleviminzi**
I also cleaned up the `read_index_select` and `write_index_insert` calls a bit by just passing in structs instead of each individual field from the struct.

### CODE REVIEW

**Comment by dleviminzi** [comment]
@asg017 Forgot all about this, but does it look alright?

**Comment by sroussey** [comment]
spelling: contents to be read into **prorgam** memory.

**Comment by dleviminzi** [comment]
> spelling: contents to be read into **prorgam** memory. 

Will fix, thank you

**Comment by tom-pollak** [comment]
any update on this? was there a reason this wasn't merged with #4

File: src/sqlite-vss.cpp [hunk 1]
```diff
@@ -479,10 +474,10 @@ static int write_index_insert(faiss::Index *index,
         sqlite3_stmt *stmt;
         char *sql = sqlite3_mprintf(
             "insert into \"%w\".\"%w_index\"(rowid, idx) values (?, ?)",
-            schema,
-            name);
+            index_vtab->schema,
+            index->name);
```
Comment (tom-pollak) [inline src/sqlite-vss.cpp:478]: should this should be `index_vtab->name`?

currently this patch breaks creating a`faiss_shadow` index.

```
SQLite Error [1]: no such table: main.`[?]_index in "insert into "main"."`[?]_index"(rowid, idx) values (?, ?)"
```

using `index_vtab->name` fixes with this change.

**Pull request closed without merging**